relay-server/src/actors/events.rs

```diff
@@ -269,36 +269,41 @@ pub struct EventProcessor {
 }
 
 impl EventProcessor {
-    #[cfg(feature = "processing")]
-    pub fn new(
-        config: Arc<Config>,
-        rate_limiter: Option<RedisRateLimiter>,
-        geoip_lookup: Option<Arc<GeoIpLookup>>,
-    ) -> Self {
+    #[inline]
+    pub fn new(config: Arc<Config>) -> Self {
         Self {
             config,
-            rate_limiter,
-            geoip_lookup,
+            #[cfg(feature = "processing")]
+            rate_limiter: None,
+            #[cfg(feature = "processing")]
+            geoip_lookup: None,
```

Comment:
These changes are great! Would it be possible to move them into a separate PR for clarity later on when tracing back the changes?